src/framework/analyses.ml

```diff
@@ -171,8 +171,11 @@ type ('d,'g,'c,'v) ctx =
 
 exception Ctx_failure of string
 (** Failure from ctx, e.g. global initializer *)
+exception Enter_func_has_no_context
+(** Tried to call ctx() on enter_func. Caught by callstring-based approaches and turned into an initial context *)
```

Comment:
Why does this need a new exception? The point of `Ctx_failure` is that there's just one exception to catch in case `ctx` cannot provide something and it's precisely used for that. This is already well-documented behavior: https://github.com/goblint/analyzer/blob/9c785b129aea4338a0d0258bd6bb88b14a331397/src/framework/analyses.ml#L162-L163

Reply:
I think it more cleanly separates the things that could go wrong. For example, if the selective context activation functor breaks at some point, we will get useful error messages here if `context` is called with `ctx.context` missing from somewhere other than the initial constraints.

Reply:
@sim642 Is this a deal breaker for you? Otherwise, I would propose to merge so it doesn't become stale.

Reply:
I'm now wondering if it'd maybe make more sense to just have a `startcontext` (like `startstate`) that defines the initial context (e.g. as an empty call string). Then there'd be no need to abuse an exception (separate or not) for a situation which isn't exceptional at all: entering the `main` function.